src/lib/VersionConflictError.js

```diff
@@ -0,0 +1,10 @@
+function VersionConflictError({ actualVersion, expectedVersion, streamName }) {
+  Error.captureStackTrace(this, this.constructor)
+  this.message = `VersionConflict: stream ${streamName} expected version ${expectedVersion} but was at version ${actualVersion}`
+  this.name = 'VersionConflictError'
+}
+
+VersionConflictError.prototype = Object.create(Error.prototype)
+VersionConflictError.prototype.constructor = VersionConflictError
```

Comment:
:thinking: Any reason you didn't reach for `class extends`? Or even `util.inherit`?

Reply:
I copied this almost verbatim from the _Practical Microservices_ book, and didn't give it a second thought.  But we could use one of those techniques if you prefer.

Reply:
I'd vote for `class extends`. I do believe is the standard for OOP inheritance nowadays.